### ISSUE TITLE
Extend app functionality to Deezer

### ISSUE THREAD

**Issue by Cwpute** [aghontpi/ad-silence#10]
Yes, this is coming from someone with almost no knowledge on how this might work.
But!
I do know that ads in Deezer are also signaled by a separate notification which could be detected by the app and subsequently silenced.

I use Deezer and i believe it's among the topmost used streaming apps! many other users might find this helpful 👀

(also happy holidays 🎄)

**Comment by aghontpi** [comment]
Thanks for the suggestion, will add it for sure. 

Happy Holidays!

**Comment by aghontpi** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [block deezer ads](https://trello.com/c/vj04S4JI/34-block-deezer-ads)

**Comment by aghontpi** [comment]
deezer is not available in my country, (even after using vpn, the app wont let me signin). I have to find a way to overcome this.

is it supported in your country? or are you using something?

**Comment by Cwpute** [comment]
Yes it is supported in my country (France). I didn't know it would be blocked in others !

**Comment by aghontpi** [comment]
added support for Deezer in #11 , closing this issue.

@Cwpute thanks for request !

**Comment by Cwpute** [comment]
Fantastic ! thank you !!

**Comment by aghontpi** [comment]
😄  yw.

 I have to test it for a few days before publishing it, so If you want to test it immediately head over [here](https://github.com/aghontpi/ad-silence/pull/11#issuecomment-1003137920).

You can also provide feedback on whether it works or not, (im limited to an emulator 😭 , it would be of great help)

**Comment by Cwpute** [comment]
Tested it just now and it doesn't work :/ the app is running in the notifications and Deezer is checked on in the settings. Not sure why it doesn't work.

**Comment by aghontpi** [comment]
I just tested it now, works as expected 🤔 

<p>
<img width="auto" height="400px" src="https://user-images.githubusercontent.com/18617846/147835520-2ab9bb9c-4962-4b94-b8ab-887fa00a50af.png"/>
<img width="auto" height="400px" src="https://user-images.githubusercontent.com/18617846/147835524-82df505a-8ae4-48d2-9fef-d242e2482608.png"/>
</p>

**Comment by aghontpi** [comment]
can you take a screenshot of the Deezer notification when an ad plays next time and post it here.

**Comment by Cwpute** [comment]
![Uploading Screenshot_20211231-203630.png…]()

**Comment by Cwpute** [comment]
![Screenshot_20211231-203630](https://user-images.githubusercontent.com/80604512/147837287-a7c8c709-5668-4928-92fd-6c8bc522c5ce.png)

**Comment by aghontpi** [comment]
thanks for the screenshot, from the notifications..

- either `ad-silence/app` did not recognise the notification or it was killed by system (but if the system has killed it, there would be no notification in the first place)

I can tell the above has happened for sure because if a `configured app` (i.e) "`Deezer/Spotify.. in preference`" posts a new notification, the  title of `ad-silence` would be "`Ad-silence . now`" but it shows "`Ad-silence . 16h`", which indicates `adsilence` last detected something 16 hours ago and did not detect Deezer or anything since. (can you test it again with the `build 10` from #11)

or the app might be killed (but notification is not removed...which has a less probability, need logcat to confirm this)

Can you specify the device model/specs? (optional, if you are into dev.. and have a pc can you take a [logcat](https://developer.android.com/studio/command-line/logcat). I can send you the instructions)

**Comment by aghontpi** [comment]
If I have extra time available, I can decomple/reverse engineer the Deezer app to patch the location restriction..then I can test it on my device, fingers crossed~

**Comment by Cwpute** [comment]
i already tried using logcat for debugging and i couldn't get it to work… i might have a better chance if someone helps me.
I don't think the app was killed because it appears in the listed open apps.
I'll try again with the new build.

**Comment by Cwpute** [comment]
My phone is a Huawei Y5 2018 DRA-L21 running on Android 8.10.
I don't think the app was killed as it was listed in the opened-apps window as Deezer was playing.
I'll try to use it with Spotify !
Last time i tried to use Logcat it didn't work… but if someone helps me through i might be able to use it.

**Comment by Cwpute** [comment]
It works perfectly with Spotify on the same device… so the issue must be specific to the Deezer app.

**Comment by aghontpi** [comment]
there should be no hiccups for android 8 and`ad-silence` would not show up in recents 🤔  (its configured not to be seen in recents screen/running apps screen). but Spotify works.. so it clears the possibility system killing the app.

What problems did you face, I can help you set it up.

**Comment by aghontpi** [comment]
Can you verify Deezer's package name is `deezer.android.app`, it will be under app Info.

**Comment by aghontpi** [comment]
An update~

I have decided it would be better to merge the Deezer branch #11 into master instead of keeping it stale, So others can test this and report.

Put Deezer under an experimental section and ship with main build. (something like below or a better UI)

![experimental-example-ui](https://user-images.githubusercontent.com/18617846/149392124-399e7047-e32f-480e-b4d4-3b9ed5128fce.jpg)

**Comment by famewolf** [comment]
I have not seen deezer in the list.  If it's hidden unless the app is installed I can install it and try to get a logcat along with some screenshots.

**Comment by aghontpi** [comment]
Closing as the new release has options for adding and configuring custom apps,  https://github.com/aghontpi/ad-silence/releases/tag/v0.7.0

Sorry it took this long to implement.

You can use a combination of debug view and add/edit custom app to implement for new apps, https://github.com/aghontpi/ad-silence?tab=readme-ov-file#custom-apps

at that time, I couldn't replicate/ detect due to location restrictions and vpn.

**Comment by aghontpi** [comment]
thanks @famewolf & @Cwpute, again sorry for the late implementation.
